packages/@aws-cdk/aws-lambda-event-sources/lib/sqs.ts

```diff
@@ -81,6 +89,7 @@ export class SqsEventSource implements lambda.IEventSource {
       enabled: this.props.enabled,
       eventSourceArn: this.queue.queueArn,
       filters: this.props.filters,
+      filterCriteria: this.props.filterCriteria,
```

Comment:
ugh looks like because `filters` is deprecated, a lot of tests need the `testDeprecated` wrapper. This is a bit of a nit, but instead of doing that (it feels slightly weird to have `testDeprecated` on a ton of test because of this change), I wonder if we could do it slightly differently...

Could we change `this.props.filters` into the type that `filterCriteria` accepts here, so that we don't have to utilize the deprecated property? i.e. something like:

```ts
const filterCriteria = this.props.filterCriteria;
filterCriteria.concat(toCriteria(this.props.filters));
```

It is usually good practice to not actually use the deprecated properties imo. They only live on for legacy sake and we should migrate off of them wherever possible.

Reply:
Now I think it's works fine, I remove `filterCriteria` and remove the deprecated tag from `filters` and use FilterCriteria class to convert the object